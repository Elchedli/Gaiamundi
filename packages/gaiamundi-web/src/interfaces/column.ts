export interface Column {
  name: string;
  sample: number;
  source: string;
  validity: string;
  isGeoCode: boolean;
}

export interface DatasetColumn extends Column {
  dataset: string;
}
