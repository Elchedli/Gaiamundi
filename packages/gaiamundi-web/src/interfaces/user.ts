import { ApiResponse } from './api';

export interface UserCredentials {
  email: string;
  password: string;
}

export interface UserSignUpFields extends UserCredentials {
  username: string;
}

export interface User extends UserSignUpFields {
  id: number;
  provider: 'local';
  confirmed: boolean;
  blocked: boolean;
  createdAt: string;
  updatedAt: string;
}

export type UserAuthResponse = {
  jwt: string;
  user: User;
};

export type ApiUserAuthResponse = ApiResponse<UserAuthResponse>;
