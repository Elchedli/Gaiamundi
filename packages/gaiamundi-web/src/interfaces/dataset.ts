import { ApiDocument } from './api';
import { UploadedFile } from './file';
import { User } from './user';

export interface DatasetBase {
  name: string;
  origin: number;
  isPublic: boolean;
}

export interface DatasetStub extends DatasetBase {
  owner: number;
  csv: number;
}

export interface Dataset extends DatasetBase {
  owner: ApiDocument<User>;
  csv: ApiDocument<UploadedFile>;
}
