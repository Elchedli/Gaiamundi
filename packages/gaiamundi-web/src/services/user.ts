import { User, UserSignUpFields } from 'interfaces/user';
import { strapi } from './strapi';

export const updateCurrentUser = async (
  userId: number,
  data: Pick<User, 'username' | 'email' | 'password' | 'Avatar'>
) => {
  return await strapi.updateCurrentUser(userId, data);
};

export const uploadUserProfileImage = async (
  userId: number,
  file: File,
  ref: string,
  filename?: string,
  refId?: number,
  field?: string
) => {
  const uploadedFile = await strapi.uploadFile(
    file,
    ref,
    filename,
    refId,
    field
  );

  return uploadedFile.url;
};

export const sendForgotPasswordEmail = async (email: string) => {
  return await strapi.forgotPassword(email);
};

export const login = async (email: string, password: string) => {
  return await strapi.login(email, password);
};

export const resetPassword = async (
  code: string,
  password: string,
  password2: string
) => {
  return await strapi.resetPassword(code, password, password2);
};

export const signUp = async (newUser: UserSignUpFields) => {
  return await strapi.register(newUser);
};

export const uploadAvatar = async (file: File, refId: number) => {
  return await strapi.uploadFile(
    file,
    'plugin::users-permissions.user',
    `${refId}_thumbnail.png`,
    refId,
    'Avatar'
  );
};
