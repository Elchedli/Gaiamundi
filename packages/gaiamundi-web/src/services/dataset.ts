import { Column } from 'interfaces/column';
import { DataFragmentStub } from 'interfaces/data-fragment';
import { DatasetStub } from 'interfaces/dataset';
import { ContentType, strapi } from './strapi';

// export const createDataset = async (data: DatasetStub) => {
//   return await strapi.create<DatasetStub>(ContentType.DATASET, data);
// };

export const addDataToPageCarto = async (
  pageCartoId: number,
  data: DatasetStub,
  selectedColumns: Column[]
) => {
  const dataset = await strapi.create<DatasetStub>(ContentType.DATASET, data);
  const columns = await Promise.all(
    selectedColumns.map((column) =>
      strapi.create<Column>(ContentType.COLUMN, column)
    )
  );
  return await strapi.create<DataFragmentStub>(ContentType.DATA_FRAGMENT, {
    columns: columns.map((column) => column.data.id),
    dataset: dataset.data.id,
    page_carto: pageCartoId,
  });
};

export const uploadCsv = async (file: File) => {
  return await strapi.uploadFile(file, 'api::dataset.dataset');
};
