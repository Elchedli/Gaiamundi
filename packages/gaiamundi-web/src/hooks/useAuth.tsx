import React from 'react';
import {
  QueryObserverResult,
  RefetchOptions,
  UseMutateFunction,
  useQuery,
  useQueryClient,
} from 'react-query';

import { Alert } from 'components/Alert/Alert';
import { LoadingMessage } from 'components/Loader/LoadingMessage';
import { User } from 'interfaces/user';
import { strapi } from '../services/strapi';
import { useLocalStorageState } from './useLocalStorageState';

// const updateAccount = async (id: number, userData: UserSignUpFields) => {
//   return (await strapi.update(ContentType.USERS, id, userData)) as User;
// };

export interface AuthContextValue {
  user: User | undefined;
  isAuthenticated: boolean;
  setUser: (data: User | undefined) => void;
  authenticate: (user: User, jwt: string) => void;
  logout: UseMutateFunction;
  refetchUser: (
    options?: RefetchOptions | undefined
  ) => Promise<QueryObserverResult<User, Error>>;
  error: Error | null;
}

const AuthContext = React.createContext<AuthContextValue | null>(null);
AuthContext.displayName = 'AuthContext';

export interface AuthProviderProps {
  children: React.ReactNode;
}

export const AuthProvider = ({ children }: AuthProviderProps): JSX.Element => {
  const { value: jwtToken, persistValue: persistJwtToken } =
    useLocalStorageState('token');
  const queryClient = useQueryClient();
  const key = 'auth-user';

  const {
    data: user,
    error,
    isLoading,
    refetch,
  } = useQuery<User, Error>({
    queryKey: key,
    queryFn: async () => {
      return await strapi.currentUser(jwtToken);
    },
    enabled: !!jwtToken,
  });

  const setUser = (data: User | undefined) =>
    queryClient.setQueryData(key, data);

  const authenticate = (user: User, jwt: string) => {
    setUser(user);
    persistJwtToken(jwt);
  };

  const logout = () => {
    persistJwtToken('');
    queryClient.clear();
  };

  if (isLoading) {
    return <LoadingMessage />;
  }

  return (
    <AuthContext.Provider
      value={{
        user,
        isAuthenticated: !!jwtToken,
        error,
        setUser,
        authenticate,
        refetchUser: refetch,
        logout,
      }}
    >
      {error && <Alert>{error.message}</Alert>}
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => {
  const context = React.useContext(AuthContext);
  if (!context) {
    throw new Error(`useAuth must be used within an AuthProvider`);
  }
  return context;
};
