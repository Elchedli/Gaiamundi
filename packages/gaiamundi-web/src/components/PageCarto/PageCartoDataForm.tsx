import { FC, useEffect, useState } from 'react';
import { UploadedFile } from 'interfaces/file';
import { useForm } from 'react-hook-form';
import { useMutation } from 'react-query';
import { ExclamationTriangleIcon } from '@heroicons/react/24/solid';

import { CsvUploader } from './CsvUploader';
import { useRequireAuth } from 'hooks/useRequireAuth';
import { useToast } from 'hooks/useToast';
import { Label } from 'components/Forms/Inputs/Label';
import { TextInput } from 'components/Forms/Inputs/TextInput';
import { ApiErrorAlert } from 'components/Alert/ApiErrorMessage';
import { ApiError } from 'interfaces/api';
import { Button } from 'components/Button/Button';
import { ToggleSwitch } from 'components/Forms/Inputs/ToggleSwitch';
import { DatasetStub } from 'interfaces/dataset';
import { createDataset } from 'services/dataset';
import { Alert } from 'components/Alert/Alert';
import { CsvColumn, DatasetColumnPicker } from './DatasetColumnPicker';

export const PageCartoDataForm: FC = () => {
  const { user } = useRequireAuth();
  const { addToast } = useToast();
  const [file, setFile] = useState<UploadedFile | undefined>();
  const [columns, setColumns] = useState<CsvColumn[]>([]);
  const {
    register,
    handleSubmit,
    formState: { errors },
    setValue,
  } = useForm<DatasetStub>();

  const { isError, error, isLoading, mutateAsync } = useMutation({
    mutationFn: async (data: DatasetStub) => {
      return await createDataset({ ...data, owner: user?.id || 0 });
    },
    onSuccess: (_response, _formData) => {
      addToast({
        title: 'Jeu de données importé',
        description: `Votre jeu de données a été crée avec succès`,
        type: 'success',
      });
    },
  });

  const handleFileUpload = (file: UploadedFile) => {
    setFile(file);
  };

  const handleFileParse = (columnNames: string[] | undefined) => {
    if (Array.isArray(columnNames)) {
      const cols = columnNames.reduce((acc, curr) => {
        let metadata: { [key: string]: string } = {};
        let name = curr;
        const [match] = name.match(/\[(.+)\]/gi) || [''];
        if (match) {
          name = name.replace(match, '');
          const meta = match.slice(1, -1);
          metadata = meta.split('&').reduce((params, param) => {
            const [key, value] = param.split('=');
            params[key] = value ? value.trim() : '';
            return params;
          }, metadata);
        }
        acc.push({
          name: name.trim(),
          source: 's' in metadata ? metadata['s'] : '',
          validity: 'v' in metadata ? metadata['v'] : '',
        });
        return acc;
      }, [] as CsvColumn[]);
      console.log(cols);
      setColumns(cols);
    }
  };

  const handleFileUploadCancel = () => {
    setFile(undefined);
  };

  const onSubmit = (data: any) => {
    mutateAsync(data);
  };

  useEffect(() => {
    setValue('name', file?.name || '');
  }, [file]);

  return (
    <div className="p-4 bg-slate-100 border border-slate-300 rounded shadow-xl">
      <CsvUploader
        onFileUploaded={handleFileUpload}
        onCancel={handleFileUploadCancel}
        onFileParse={handleFileParse}
      />
      {file && (
        <form onSubmit={handleSubmit(onSubmit)}>
          {isError && <ApiErrorAlert error={error as ApiError} />}
          <div className="flex flex-col">
            <div className="mt-3">
              <Label htmlFor="name">Nom du jeu de données</Label>
              <TextInput
                id="name"
                className="w-full"
                {...register('name', {
                  required: `Veuillez saisir le nom de votre jeu de données`,
                })}
              />
              {errors.name && (
                <div className="mt-2 text-xs text-red-600">
                  {errors.name.message}
                </div>
              )}
            </div>
            <div className="mt-3">
              <Label htmlFor="origin">Origine du fichier</Label>
              <TextInput
                id="origin"
                className="w-full"
                {...register('origin', {
                  required: `Veuillez saisir l'origine de votre fichier de données`,
                })}
              />
              {errors.origin && (
                <div className="mt-2 text-xs text-red-600">
                  {errors.origin.message}
                </div>
              )}
            </div>
            <div className="mt-3">
              <ToggleSwitch
                label={'Accès Public'}
                defaultChecked={false}
                onChange={(v) => {
                  setValue('isPublic', v);
                }}
              />
              <Alert
                className="p-2 text-xs"
                type="warning"
                icon={ExclamationTriangleIcon}
              >
                Attention: Risque de non conformité <br />
                <small>
                  (Origine du fichier, source et validité des colonnes)
                </small>
              </Alert>
            </div>
            <div>
              <TextInput
                id="csv"
                type={'hidden'}
                {...register('csv', {
                  required: 'Le fichier CSV est obligatoire',
                })}
              />
            </div>
            <div>
              <DatasetColumnPicker columns={columns} />
            </div>
            <div className="flex justify-end">
              <Button
                type="submit"
                size="lg"
                className="inline-flex justify-center"
                isLoading={isLoading}
              >
                Valider
              </Button>
            </div>
          </div>
        </form>
      )}
    </div>
  );
};
