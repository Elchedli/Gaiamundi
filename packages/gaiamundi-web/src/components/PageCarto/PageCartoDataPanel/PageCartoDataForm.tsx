import { ExclamationTriangleIcon } from '@heroicons/react/24/solid';
import { UploadedFile } from 'interfaces/file';
import { FC, useState } from 'react';
import { Controller, useForm } from 'react-hook-form';
import { useMutation, useQueryClient } from 'react-query';

import { Alert } from 'components/Alert/Alert';
import { ApiErrorAlert } from 'components/Alert/ApiErrorMessage';
import { Button } from 'components/Button/Button';
import { Label } from 'components/Forms/Inputs/Label';
import { TextInput } from 'components/Forms/Inputs/TextInput';
import { ToggleSwitch } from 'components/Forms/Inputs/ToggleSwitch';
import { useRequireAuth } from 'hooks/useRequireAuth';
import { useToast } from 'hooks/useToast';
import { ApiError } from 'interfaces/api';
import { Column } from 'interfaces/column';
import { DatasetStub } from 'interfaces/dataset';
import { addDataToPageCarto } from 'services/dataset';
import { CsvUploader } from './CsvUploader';
import { DatasetColumnPicker } from './DatasetColumnPicker';

type FormData = DatasetStub & {
  fragmentName: string;
  columns: Column[];
};

type PageCartoDataFormProps = {
  pageCartoId: number;
  onSubmit: (dataForm: FormData) => void;
};

export const PageCartoDataForm: FC<PageCartoDataFormProps> = ({
  pageCartoId,
  onSubmit,
}) => {
  const queryClient = useQueryClient();
  const { user } = useRequireAuth();
  const [columns, setColumns] = useState<Column[]>([]);
  const { addToast } = useToast();
  const {
    register,
    handleSubmit,
    formState: { errors, isValid },
    setValue,
    control,
  } = useForm<FormData>();

  const { isError, error, isLoading, mutateAsync } = useMutation({
    mutationFn: async ({ columns, fragmentName, ...dataset }: FormData) => {
      return await addDataToPageCarto(
        pageCartoId,
        fragmentName,
        { ...dataset, owner: user?.id || 0 },
        columns
      );
    },
    onSuccess: (_response, formData) => {
      addToast({
        title: 'Jeu de données importé',
        description: `Votre jeu de données a été crée avec succès`,
        type: 'success',
      });
      queryClient.invalidateQueries({ queryKey: ['page-carto', pageCartoId] });
      onSubmit(formData);
    },
  });

  const handleFileUpload = (file: UploadedFile) => {
    setValue('name', file?.name || '');
    setValue('fragmentName', file?.name || '');
  };

  const handleFileCancel = () => {
    setValue('name', '');
    setValue('fragmentName', '');
    setColumns([]);
  };

  const handleFileParse = (columns: Column[]) => {
    setColumns(columns);
  };

  const onSubmitForm = (data: FormData) => {
    mutateAsync(data);
  };

  return (
    <div className="p-4 bg-slate-100 border border-slate-300 rounded shadow-xl">
      <form onSubmit={handleSubmit(onSubmitForm)}>
        {isError && <ApiErrorAlert error={error as ApiError} />}
        <div className="flex flex-row">
          <div className="flex flex-col w-1/2 p-2">
            <div className="mt-3">
              <Label htmlFor="name">
                1. Uploader des données à partir d&apos;un fichier CSV
              </Label>
              <Controller
                name="csv"
                control={control}
                defaultValue={undefined}
                rules={{
                  required: 'Le fichier CSV est obligatoire',
                }}
                render={({ field }) => {
                  return (
                    <CsvUploader
                      onUpload={handleFileUpload}
                      onParse={handleFileParse}
                      onCancel={handleFileCancel}
                      {...field}
                    />
                  );
                }}
              />
              {errors.csv && (
                <div className="mt-2 text-xs text-red-600">
                  {errors.csv.message}
                </div>
              )}
            </div>
            <div className="mt-3">
              <Label>2. Sélectionnez les colonnes à utiliser :</Label>
              <Controller
                name="columns"
                control={control}
                defaultValue={undefined}
                rules={{
                  required:
                    'Vous devez sélectionner les colonnes ainsi que le Géo Code.',
                  validate: (cols) => {
                    return cols.length > 0 && cols.some((c) => c.isGeoCode);
                  },
                }}
                render={({ field }) => {
                  return <DatasetColumnPicker data={columns} {...field} />;
                }}
              />
            </div>
          </div>
          <div className="flex flex-col w-1/2 p-2">
            <div className="mt-3">
              <Label htmlFor="name">Nom du jeu de données</Label>
              <TextInput
                id="fragmentName"
                className="w-full"
                {...register('fragmentName', {
                  required: `Veuillez saisir le nom de votre jeu de données`,
                })}
              />
              {errors.fragmentName && (
                <div className="mt-2 text-xs text-red-600">
                  {errors.fragmentName.message}
                </div>
              )}
            </div>
            <div className="mt-3">
              <Label htmlFor="origin">Origine du fichier</Label>
              <TextInput
                id="origin"
                className="w-full"
                {...register('origin', {
                  required: `Veuillez saisir l'origine de votre fichier de données`,
                })}
              />
              {errors.origin && (
                <div className="mt-2 text-xs text-red-600">
                  {errors.origin.message}
                </div>
              )}
            </div>
            <div className="mt-3">
              <ToggleSwitch
                label={'Accès Public'}
                defaultChecked={false}
                onChange={(v) => {
                  setValue('isPublic', v);
                }}
              />
              <Alert
                className="p-2 text-xs"
                type="warning"
                icon={ExclamationTriangleIcon}
              >
                Attention: Risque de non conformité <br />
                <small>
                  (Origine du fichier, source et validité des colonnes)
                </small>
              </Alert>
            </div>
            <div className="flex justify-end mt-3">
              <Button
                type="submit"
                size="lg"
                isLoading={isLoading}
                disabled={!isValid}
              >
                Valider
              </Button>
            </div>
          </div>
        </div>
      </form>
    </div>
  );
};
