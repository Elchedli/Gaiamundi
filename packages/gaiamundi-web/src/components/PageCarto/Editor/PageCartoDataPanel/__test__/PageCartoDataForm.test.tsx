import { render, screen } from '@testing-library/react';
import { useAuth } from 'hooks/useAuth';
import { Column } from 'interfaces/column';
import { DatasetStub } from 'interfaces/dataset';
import { User } from 'interfaces/user';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { useMutation } from 'react-query';
import { MemoryRouter as Router } from 'react-router-dom';
import { PageCartoDataForm } from '../PageCartoDataForm';

const server = setupServer(
  rest.post('/add-data-to-page-carto', (req, res, ctx) => {
    return res(ctx.status(201), ctx.json({ success: true }));
  })
);

jest.mock('react-query', () => ({
  ...jest.requireActual('react-query'),
  useMutation: jest.fn(),
  useQueryClient: jest.fn(),
}));

const mockUseMutation = useMutation as jest.Mock;

const user: User = {
  id: 1,
  confirmed: true,
  blocked: false,
  username: 'testuser',
  email: 'testuser@test.com',
  provider: 'local',
  password: 'testpass',
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

jest.mock('hooks/useAuth');
jest.mock('react-query', () => ({
  useMutation: jest.fn(),
  useQueryClient: jest.fn(),
}));

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

describe('PageCartoDataForm', () => {
  beforeEach(() => {
    (useAuth as jest.Mock).mockImplementation(() => ({
      isAuthenticated: true,
      user: user,
    }));
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders the correct text elements', async () => {
    mockUseMutation.mockReturnValue({
      invoke: jest.fn(),
      result: { loading: false },
      reset: jest.fn(),
    });

    const { container } = render(
      <Router>
        <PageCartoDataForm
          pageCartoId={0}
          onSubmit={function (
            // eslint-disable-next-line @typescript-eslint/no-unused-vars
            dataForm: DatasetStub & { fragmentName: string; columns: Column[] }
          ): void {
            throw new Error('Function not implemented.');
          }}
        />
      </Router>
    );

    expect(
      screen.getByText("1. Uploader des données à partir d'un fichier CSV")
    ).toBeInTheDocument();
    expect(screen.getByText('Nom du jeu de données')).toBeInTheDocument();
    expect(
      screen.getByText('Attention: Risque de non conformité')
    ).toBeInTheDocument();
    expect(screen.getByText('Valider')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
