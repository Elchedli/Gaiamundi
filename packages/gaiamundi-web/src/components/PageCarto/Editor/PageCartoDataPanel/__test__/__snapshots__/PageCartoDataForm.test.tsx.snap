// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCartoDataForm renders the correct text elements 1`] = `
<div>
  <div
    class="p-4 bg-slate-100 border border-slate-300 rounded shadow-xl"
  >
    <form>
      <div
        class="flex flex-row"
      >
        <div
          class="flex flex-col w-1/2 p-2"
        >
          <div
            class="mt-3"
          >
            <label
              class="text-sm font-medium text-gray-900 dark:text-gray-300"
              for="name"
            >
              1. Uploader des données à partir d'un fichier CSV
            </label>
            <div
              class="self-center"
            >
              <div
                data-testid="dropzone"
              >
                <label
                  class="text-sm font-medium text-gray-900 dark:text-gray-300 flex flex-col border-2 border-gray-300 border-dashed rounded-lg cursor-pointer bg-gray-50 dark:hover:bg-bray-800 dark:bg-gray-700 hover:bg-gray-100 dark:border-gray-600 dark:hover:border-gray-500 dark:hover:bg-gray-600"
                  for="dropzone-file"
                >
                  <div
                    class="flex flex-col items-center justify-center pt-5 pb-6"
                  >
                    <svg
                      aria-hidden="true"
                      class="w-10 h-10 mb-3 text-gray-400"
                      fill="none"
                      stroke="currentColor"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                    <p
                      class="mb-2 text-sm text-gray-500 dark:text-gray-400"
                    >
                      <span>
                        <span
                          class="font-semibold"
                        >
                          Cliquer pour ajouter 
                        </span>
                        ou glisser et déposer
                      </span>
                    </p>
                  </div>
                  <div>
                    <div
                      class="relative"
                    >
                      <input
                        aria-describedby="file_upload_help"
                        class="border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500 rounded-lg p-2.5 text-sm hidden"
                        data-testid="hidden-file-input"
                        id="dropzone-file"
                        type="file"
                      />
                    </div>
                  </div>
                </label>
              </div>
            </div>
          </div>
          <div
            class="mt-3"
          >
            <label
              class="text-sm font-medium text-gray-900 dark:text-gray-300"
            >
              2. Sélectionnez les colonnes à utiliser :
            </label>
            <div
              class="border-slate-300"
            >
              <div>
                <div
                  class="relative"
                >
                  <div
                    class="pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-gray-500 dark:text-gray-400"
                      fill="currentColor"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10.5 3.75a6.75 6.75 0 100 13.5 6.75 6.75 0 000-13.5zM2.25 10.5a8.25 8.25 0 1114.59 5.28l4.69 4.69a.75.75 0 11-1.06 1.06l-4.69-4.69A8.25 8.25 0 012.25 10.5z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                  <input
                    autocomplete="false"
                    class="border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500 pl-10 rounded-lg p-2.5 text-sm w-full mb-2"
                    id="keywords"
                    value=""
                  />
                </div>
              </div>
              <div
                aria-colcount="5"
                aria-multiselectable="true"
                aria-rowcount="1"
                class="rdg r104f42s7-0-0-beta-22 fill-grid"
                dir="ltr"
                role="grid"
                style="grid-template-rows: 35px; --rdg-header-row-height: 35px; --rdg-summary-row-height: 35px; --rdg-sign: 1; grid-template-columns: 35px 80px auto 80px 80px; --rdg-frozen-left-0: 0px; --rdg-frozen-left-1: 35px;"
              >
                <div
                  aria-rowindex="1"
                  class="rdg-header-row h197vzie7-0-0-beta-22"
                  role="row"
                  style="--rdg-grid-row-start: 1;"
                >
                  <div
                    aria-colindex="1"
                    aria-selected="false"
                    class="rdg-cell c1wupbe7-0-0-beta-22 rdg-cell-frozen cd0kgiy7-0-0-beta-22"
                    role="columnheader"
                    style="grid-column-start: 1; inset-inline-start: var(--rdg-frozen-left-0);"
                    tabindex="0"
                  >
                    <label
                      class="rdg-checkbox-label c8rwwhf7-0-0-beta-22"
                    >
                      <input
                        aria-label="Select All"
                        class="rdg-checkbox-input c1rb4brs7-0-0-beta-22"
                        tabindex="-1"
                        type="checkbox"
                      />
                      <div
                        class="rdg-checkbox c1bvvwcc7-0-0-beta-22"
                      />
                    </label>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-selected="false"
                    class="rdg-cell c1wupbe7-0-0-beta-22 rdg-cell-resizable celq7o97-0-0-beta-22 rdg-cell-frozen cd0kgiy7-0-0-beta-22 rdg-cell-frozen-last c1730fa47-0-0-beta-22"
                    role="columnheader"
                    style="grid-column-start: 2; inset-inline-start: var(--rdg-frozen-left-1);"
                    tabindex="-1"
                  >
                    GéoCode
                  </div>
                  <div
                    aria-colindex="3"
                    aria-selected="false"
                    class="rdg-cell c1wupbe7-0-0-beta-22 rdg-cell-resizable celq7o97-0-0-beta-22"
                    role="columnheader"
                    style="grid-column-start: 3;"
                    tabindex="-1"
                  >
                    Colonne
                  </div>
                </div>
                <div
                  class="m1l09lto7-0-0-beta-22"
                  data-measuring-cell-key="select-row"
                  style="grid-column-start: 1; min-width: 35px; max-width: 35px;"
                />
                <div
                  class="m1l09lto7-0-0-beta-22"
                  data-measuring-cell-key="isGeoCode"
                  style="grid-column-start: 2; min-width: 80px; max-width: 80px;"
                />
                <div
                  class="m1l09lto7-0-0-beta-22"
                  data-measuring-cell-key="name"
                  style="grid-column-start: 3; min-width: 80px;"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col w-1/2 p-2"
        >
          <div
            class="mt-3"
          >
            <label
              class="text-sm font-medium text-gray-900 dark:text-gray-300"
              for="name"
            >
              Nom du jeu de données
            </label>
            <div>
              <div
                class="relative"
              >
                <input
                  class="border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500 rounded-lg p-2.5 text-sm w-full"
                  id="fragmentName"
                  name="fragmentName"
                />
              </div>
            </div>
          </div>
          <div
            class="mt-3"
          >
            <label
              class="text-sm font-medium text-gray-900 dark:text-gray-300"
              for="origin"
            >
              Origine du fichier
            </label>
            <div>
              <div
                class="relative"
              >
                <input
                  class="border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500 rounded-lg p-2.5 text-sm w-full"
                  id="origin"
                  name="origin"
                />
              </div>
            </div>
          </div>
          <div
            class="mt-3"
          >
            <div
              class="flex items-center"
            >
              <button
                aria-checked="false"
                aria-labelledby="headlessui-label-:r1:"
                class="bg-gray-200 relative inline-flex h-6 w-11 items-center rounded-full transition-colors focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
                data-headlessui-state=""
                data-testid="toggle-switch"
                id="headlessui-switch-:r0:"
                role="switch"
                tabindex="0"
                type="button"
              >
                <span
                  class="translate-x-1 inline-block h-4 w-4 transform rounded-full bg-white transition-transform"
                />
              </button>
              <label
                class="ml-4 cursor-pointer"
                id="headlessui-label-:r1:"
              >
                Accès Public
              </label>
            </div>
            <div
              class="p-3 my-3 rounded border border-orange-500 bg-orange-200 text-orange-700 p-2 text-xs"
              role="alert"
            >
              <div
                class="flex flex-row"
              >
                <svg
                  aria-hidden="true"
                  class="mr-2"
                  fill="currentColor"
                  height="24"
                  stroke="current"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M9.401 3.003c1.155-2 4.043-2 5.197 0l7.355 12.748c1.154 2-.29 4.5-2.599 4.5H4.645c-2.309 0-3.752-2.5-2.598-4.5L9.4 3.003zM12 8.25a.75.75 0 01.75.75v3.75a.75.75 0 01-1.5 0V9a.75.75 0 01.75-.75zm0 8.25a.75.75 0 100-1.5.75.75 0 000 1.5z"
                    fill-rule="evenodd"
                  />
                </svg>
                <div>
                  Attention: Risque de non conformité 
                  <br />
                  <small>
                    (Origine du fichier, source et validité des colonnes)
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex justify-end mt-3"
          >
            <button
              class="group inline-flex h-min w-fit items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg text-white bg-blue-700 border border-transparent hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 disabled:hover:bg-blue-700 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 dark:disabled:hover:bg-blue-600 cursor-not-allowed opacity-50"
              data-testid="button-element"
              disabled=""
              type="submit"
            >
              <span
                class="flex items-center text-base px-5 py-2.5 rounded-md"
              >
                Valider
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
